node/src/database/db_initializer.rs

```diff
@@ -1033,7 +1157,175 @@ mod tests {
         }
         let subject = DbInitializerReal::default();
 
-        let _ = subject.initialize(&home_dir, true, MigratorConfig::panic_on_migration());
+        let _ = subject.initialize(
+            &home_dir,
+            true,
+            DbInitializationConfig::panic_on_migration(),
+        );
+    }
+
+    fn connection_special_setup_changed_assertion<F, E>(init_conn: Option<&Connection>, modifier: F)
+    where
+        F: FnOnce() -> Option<Box<dyn ConnectionWrapper>>,
+        E: Debug,
+    {
+        fn ask_pragma(conn: &Connection) -> u16 {
+            conn.pragma_query_value(None, "busy_timeout", |row| row.get::<usize, u16>(0))
+                .unwrap()
+        }
+        fn ending_assertion(conn: &Connection, value_by_default: u16) {
+            let after = ask_pragma(conn.borrow());
+            assert_eq!(value_by_default, 5000);
```

Comment:
Can you move this assertion out from between the two parts of the `after` assertion?